towhee/hub/hub_tools.py

```diff
@@ -423,47 +425,74 @@ def create_repo(repo: str, token: str, repo_type: str) -> None:
+    with open(new_file, 'w', encoding='utf-8') as f2:
+        f2.write(file_text)
 
-    First clone the repo, then download and rename the template repo file.
 
+def init_file_structure(author: str, repo: str, repo_type: str, tag: str) -> None:
+    """
+    Initialize the file structure with template.
+    First clone the repo, then initialize it with the template file structure.
     Args:
         author (`str`):
             The account name.
         repo (`str`):
             The repo name.
         repo_type (`str`):
-            Which category of repo to create, only one can be used, includes
-            ('model', 'operator', 'pipeline', 'dataset').
-
+            Which category of repo to create, only one can be used, includes ('operator', 'pipeline').
+        tag (`str`):
+            The tag of the repo.
     Raises:
         (`HTTPError`)
             Raise error in request.
         (`OSError`)
             Raise error in writing file.
     """
-    links = 'https://hub.towhee.io/' + author + '/' + repo + '.git'
-    subprocess.call(['git', 'clone', links])
-    repo_file_name = repo.replace('-', '_')
+    repo_temp = 'pipeline-template'
+    nn_framework = ''
     if repo_type == 'operator':
-        # TODO: distinguish nnop and pyop (Shiyu)
-        lfs_files = obtain_lfs_extensions('towhee', 'operator-template', 'main')
-        commit = latest_branch_commit('towhee', 'operator-template', 'main')
-        file_list = get_file_list('towhee', 'operator-template', commit)
-        download_files('towhee', 'operator-template', 'main', file_list, lfs_files, str(Path.cwd() / repo), False)
-
-        (Path(repo) / 'operator_template.py').rename(Path(repo) / (repo_file_name + '.py'))
-        (Path(repo) / 'operator_template.yaml').rename(Path(repo) / (repo_file_name + '.yaml'))
-
-    elif repo_type == 'pipeline':
-        lfs_files = obtain_lfs_extensions('towhee', 'pipeline-template', 'main')
-        commit = latest_branch_commit('towhee', 'pipeline-template', 'main')
-        file_list = get_file_list('towhee', 'pipeline-template', commit)
-        download_files('towhee', 'pipeline-template', 'main', file_list, lfs_files, str(Path.cwd() / repo), False)
-
-        (Path(repo) / 'pipeline_template.yaml').rename(Path(repo) / (repo_file_name + '.yaml'))
+        repo_temp = 'pyoperator-template'
+        op_choice = input('Is it NNOperator (there are neural network or model related codes)? '
+                          '[Y|n, press Enter will default to \'No\']  ')
+        if op_choice.lower() in ['yes', 'y']:
+            nn_framework = input('What\'s the framework of your model? [press Enter will default to \'pytorch\']  ')
+            repo_temp = 'nnoperator-template'
+
+    print('Clone the repo and initialize it with template...')
+    url = f'https://towhee.io/{author}/{repo}.git'
+    url_temp = f'https://towhee.io/towhee/{repo_temp}.git'
+    git.Repo.clone_from(url=url, to_path=repo, branch=tag)
```

Comment:
im thinking that this functionality should be added in a seperate tool to avoid requiring git for all towhee users.

Reply:
Actually, the hub tool is developed for contributors, so they have to install git and git-lfs, of course the README will state that.